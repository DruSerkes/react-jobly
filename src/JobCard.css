.JobCard {
	margin: 1rem auto;
	background-color: white;
	width: 50%;
	box-shadow: 1px 2px 3px lightgray;
	padding: 1rem;
	border-radius: 1px;
	position: relative;
	text-align: right;
}

.JobCard-Info {
	text-align: left;
}

.JobCard ul {
	padding: 0;
}
.JobCard li {
	list-style-type: none;
	padding: 0;
}

.JobCard-Apply {
	position: relative;
	right: 0px;
}

.JobCard button {
	background-color: #dc3545;
	border: none;
	margin: 1rem;
	padding: 0.5rem .85rem;
	outline: none;
	border-radius: 5px;
	font-weight: 700;
	text-transform: uppercase;
	color: white;
	font-size: 1rem;
	line-height: 1.5rem;
}
