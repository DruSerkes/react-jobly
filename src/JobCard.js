import React from 'react';

const JobCard = ({ job, handleApply }) => {
	return (
		<div className="JobCard">
			<h3>{job.title}</h3>
			<ul>
				<li>{job.salary}</li>
				<li>{job.equity}</li>
			</ul>
			{/* add onclick={handleApply} */}
			<button className="JobCard-Apply">Apply</button>
		</div>
	);
};

export default JobCard;
