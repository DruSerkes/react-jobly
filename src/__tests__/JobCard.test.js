import React from 'react';
import { render } from '@testing-library/react';
import JobCard from '../JobCard';

describe('JobCard tests', () => {
	const mockJob = {
		title  : 'tester',
		salary : 120000,
		equity : 0.5
	};

	it('renders without breaking', () => {
		render(<JobCard job={mockJob} />);
	});

	it('matches snapshot', () => {
		const { asFragment } = render(<JobCard job={mockJob} />);
		expect(asFragment()).toMatchSnapshot();
	});

	it('contains job info', () => {
		const { getByText } = render(<JobCard job={mockJob} />);
		const title = getByText(mockJob.title);
		const salary = getByText(`Salary: ${mockJob.salary}`);
		const equity = getByText(`Equity: ${mockJob.equity}`);
		expect(title).toBeInTheDocument();
		expect(salary).toBeInTheDocument();
		expect(equity).toBeInTheDocument();
	});
});
