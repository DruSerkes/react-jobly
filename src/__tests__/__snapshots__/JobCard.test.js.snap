// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobCard tests matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="JobCard"
  >
    <div
      class="JobCard-Info"
    >
      <h3>
        tester
      </h3>
      <ul>
        <li>
          Salary: 120000
        </li>
        <li>
          Equity: 0.5
        </li>
      </ul>
    </div>
    <button
      class="JobCard-Apply"
    >
      Apply
    </button>
  </div>
</DocumentFragment>
`;
